MvvmCross/IoC/MvxTypeExtensions.cs

```diff
@@ -239,5 +239,54 @@ public static object CreateDefault(this Type type)
 
             return Activator.CreateInstance(type);
         }
+        
+        public static ConstructorInfo FindApplicableConstructor(this Type type, IDictionary<string, object> arguments)
+        {
+            var constructors = type.GetConstructors();
+            if (arguments == null || arguments.Count == 0)
+            {
+                return constructors.FirstOrDefault();
+            }
+            
+            var names = arguments.Keys;
+
+            foreach (var constructor in constructors)
+            {
+                var parameters = constructor.GetParameters();
```

Comment:
Even if we have the names of the arguments, we should be matching on type of arguments first

Reply:
Alright, fixed that